src/binaryen-c.h

```diff
@@ -103,6 +103,7 @@ typedef void* BinaryenFunctionTypeRef;
 // Add a new function type. This is thread-safe.
 // Note: name can be NULL, in which case we auto-generate a name
 BinaryenFunctionTypeRef BinaryenAddFunctionType(BinaryenModuleRef module, const char* name, BinaryenType result, BinaryenType* paramTypes, BinaryenIndex numParams);
+BinaryenFunctionTypeRef BinaryenGetFunctionType(BinaryenModuleRef module, BinaryenType result, BinaryenType* paramTypes, BinaryenIndex numParams);
```

Comment:
I see. So if we had an optimizer pass that does this, that would be good enough? Sounds like it would make all this simpler, avoiding adding a new API.

We need to add such an optimizer pass anyhow. Should be easy to write. I can do that unless you want to.